Assets/Character/AaronTestAssets/PlayerMovement.cs

```diff
@@ -0,0 +1,116 @@
+﻿using UnityEngine;
+using System.Collections;
+
+
+
+public class PlayerMovement : MonoBehaviour {
+
+
+    public float move_speed = 2f;
+    public float jump_power = 8f;
+    public float roll_speed = 4f;
+    public float roll_time = 0.5f;
+
+
+    private Animator animator;
+    private Rigidbody2D rigidbody;
+    private int directionFacing = 1;
+    //private ArrayList lockStates = new ArrayList();
+    private AnimatorStateInfo stateInfo;
+    private Vector3 scale;
+    private float timer;
+	
+
+	void Start () {
+
+        animator = GetComponent<Animator>();
+        rigidbody = GetComponent<Rigidbody2D>();
+       // lockStates.Add();
+        stateInfo = animator.GetCurrentAnimatorStateInfo(0);
+        scale = transform.localScale;
+
+    }
+	
+	void FixedUpdate () {
+        if (!animator.GetBool("Rolling"))
```

Comment:
Since states are exclusive, there are better ways to control them other than big if/else if/else. You can do a class based approach, Map<someStateIdentifier, stateHandler> where stateHandler would be RollState, etc. You can also see how christian did it in his branch.